### PR TITLE
tree-wide: use free_and_strdup()

### PR DESCRIPTION
**Pull request by zonque**
Use free_and_strdup() where appropriate and replace equivalent,
open-coded versions.

### CODE REVIEW

**Comment by dvdhrm** [comment]
Btw., there are a lot more cases like this:

``` c
t = strdup(foobar);
if (!t)
        return -ENOMEM;
free(storage);
storage = t;
```

In those situations, the helper is much more useful.

Anyway, the patch looks good to me. It does change semantics, though. We no longer free the storage on ENOMEM. Shouldn't be an issue in the fixed cases, though.

**Comment by zonque** [comment]
Yeah, I agree on both points you mention. We can still fix the other ones later ...

File: src/run/run.c [hunk 1]
```diff
@@ -1129,13 +1129,9 @@ int main(int argc, char* argv[]) {
...
 
                 if (arg_unit && isempty(description)) {
-                        free(description);
-                        description = strdup(arg_unit);
-
-                        if (!description) {
-                                r = log_oom();
+                        r = free_and_strdup(&description, arg_unit);
+                        if (r < 0)
                                 goto finish;
```
Comment (dvdhrm) [inline src/run/run.c:1134]: You drop `log_oom()` here, but I guess that's fine.

Comment (dvdhrm) [inline src/run/run.c:1134]: Feel free to push a fix for this directly.

**Pull request merged**